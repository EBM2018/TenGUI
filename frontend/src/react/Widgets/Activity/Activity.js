import React from 'react';

// import PropTypes from 'prop-types';
import PropTypes from 'prop-types';
import ActTextField from './ActTextField';
import ActTextCheck from './ActTextCheck';
import ActTextCheckMult from './ActTextCheckMult';
import { postFishtankInteraction } from '../../../service/API/interactions';
// import index from '../../../service/Websockets';

export default class Activity extends React.PureComponent {
    static propTypes = {
      fishtankId: PropTypes.number.isRequired,
      idInteractions: PropTypes.arrayOf([
        PropTypes.string,
        PropTypes.number,
      ]).isRequired,
      data: PropTypes.arrayOf([
        PropTypes.string,
        PropTypes.number,
      ]).isRequired,
      deleteData: PropTypes.func.isRequired,
    }

    state = {
      userReponse: {
        referenceInteractionId: 0,
        content: [
        ],
      },
    };

    componentWillMount() {
      this.genereateJSONUserResponse();
    }

    componentDidUpdate(oldProps) {
      const newProps = this.props;
      if (oldProps.data !== newProps.data) {
        this.genereateJSONUserResponse();
      }
    }

    genereateJSONUserResponse = () => {
      const { data, fishtankId } = this.props;
      const newUserResponse = {
        referenceInteractionId: 0,
        content: [
        ],
      };
      newUserResponse.referenceInteractionId = fishtankId;
      for (let i = 0; i < data.Question.length; i += 1) {
        const modDataOneResponse = {
          id: 0,
          answer: '',
        };
        modDataOneResponse.id = data.Question[i].idQuestion;
        newUserResponse.content.push(modDataOneResponse);
      }
      this.setState({ userReponse: newUserResponse });
    }

    send = () => { // pour des test Ã  la con
      const { userReponse } = this.state;
      const { fishtankId, idInteractions, deleteData } = this.props;
      console.log(userReponse);
      postFishtankInteraction(
        fishtankId,
        idInteractions.PARTICIPANT.FEEDBACK_SUBMIT,
        userReponse,
      );
      deleteData();
    };

    editResponse = (indexQuestion, newResponse) => {
      const { userReponse } = this.state;
      userReponse.content[indexQuestion].answer = newResponse;
      this.setState({ userReponse });
      // this.forceUpdate();
    };

    render() {
      const { data } = this.props;
      return (
        <div className="columns is-mobile">
          <div className="column is-9 is-offset-1">
            <h1 className="title is-t1">{data.name}</h1>
            {data.Question.map((ques, index) => {
              if (ques.type === 'field') {
                return (
                  <ActTextField
                    key={ques.idQuestion}
                    id={index}
                    text={ques.text}
                    editResponse={this.editResponse}
                  />
                );
              }
              if (ques.type === 'check') {
                return (
                  <ActTextCheck
                    key={ques.idQuestion}
                    id={index}
                    text={ques.text}
                    listTextResponse={ques.response}
                    editResponse={this.editResponse}
                  />
                );
              }
              if (ques.type === 'checkMult') {
                return (
                  <ActTextCheckMult
                    key={ques.idQuestion}
                    id={index}
                    text={ques.text}
                    listTextResponse={ques.response}
                    editResponse={this.editResponse}
                  />
                );
              }
              return (<></>);
            })}
            <div>
              <button type="submit" onClick={this.send}>
                      Send
              </button>
            </div>
          </div>
        </div>
      );
    }
}
