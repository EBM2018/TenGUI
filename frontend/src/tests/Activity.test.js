import React from 'react';
import renderer from 'react-test-renderer';
import Activity from '../react/Widgets/Activity/Activity';


test('does this render', () => {
  const component = renderer
    .create(<Activity />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
