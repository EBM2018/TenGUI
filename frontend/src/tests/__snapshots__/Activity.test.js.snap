// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does this render 1`] = `
<div>
  Questionnaire_name
  <div>
    Qu'est ce que tu as appris ?
    <div>
      <input
        id="input_response"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div>
    Qu'est ce qui t'as étonné ?
    <div>
      <input
        id="input_response"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div>
    Qu'est ce que tu voudrais avoir plus ou en plus ?
    <div>
      <input
        id="input_response"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div>
    Qu'est ce que tu voudrais avoir moins ou en moins ?
    <div>
      <input
        id="input_response"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    id="container reponse"
  >
    <div>
      Cb tu notes ce cour ?
    </div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    1
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    2
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    3
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    4
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    5
  </div>
  <div>
    <button
      onClick={[Function]}
      type="submit"
    >
      Send
    </button>
  </div>
</div>
`;
